### ISSUE TITLE
Bug: "import/*" not applied

### ISSUE THREAD

**Issue by simlu** [eslint/eslint#12150]
**Tell us about your environment**

* **ESLint Version:** ^6.2.1
* **Node Version:** v10.16.3
* **yarn Version:** 1.15.2

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "root": true,
  "extends": ["airbnb-base"],
  "env": {"es6": true, "node": true},
  "plugins": []
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

[test-project.zip](https://github.com/eslint/eslint/files/3531203/test-project.zip)

Run `. debug.sh` (tested on Ubuntu 16.04)

**What did you expect to happen?**

Error report for rule `import/no-useless-path-segments` since `index.js` is in violation and the rule shows as active (see generated debug.json)

**What actually happened? Please include the actual, raw output from ESLint.**

Only a console warning was generated. The rule shows as active but seem to not have been run

**Are you willing to submit a pull request to fix this bug?**

I'm honestly not involved enough in eslint to even understand what is going on here. However if this is an easy fix I'd be very happy to do a PR.

**Comment by mdjermanovic** [comment]
Hi @simlu,

This looks like an issue with  `eslint-plugin-import`, not eslint. While at the subject, by taking a quick look at the [test file](https://github.com/benmosher/eslint-plugin-import/blob/7ffbf03fc832df41c4e90ffd27fee43de3685224/tests/src/rules/no-useless-path-segments.js#L40) it seems that you would need to enable `"commonjs"` rule's option.

It worked for me, but as it isn't a documented option I'd recommend you to open an issue in [eslint-plugin-import](https://github.com/benmosher/eslint-plugin-import)

**Comment by ljharb** [comment]
@mdjermanovic https://github.com/airbnb/javascript/issues/2077 has already been filed.

@simlu the airbnb config assumes that everything is a babel-transpiled ES Module; I missed that you were using `require`. if enabling the `commonjs` option makes it work for you, we can reopen https://github.com/airbnb/javascript/issues/2077 to discuss enabling that.

**Comment by ljharb** [comment]
Separately, please do file an issue or PR to document the option :-)

**Comment by simlu** [comment]
Thank you very much for the reply @mdjermanovic

I still can not get the rule to fire using my test-project example. I tried the following config

```json
{
  "root": true,
  "extends": ["airbnb-base"],
  "env": {"es6": true, "node": true, "commonjs": true},
  "plugins": []
}
```

and get

```txt
/home/vinc/tmp/test-project/src/index.js
  3:1  warning  Unexpected console statement  no-console

✖ 1 problem (0 errors, 1 warning)
```


Would you mind posting your configuration?

Cheers, L~

**Comment by ljharb** [comment]
@simlu the `commonjs` option is for the rule, not for env.

**Comment by simlu** [comment]
@ljharb Would you mind telling me how to pass an option into a rule? I'm not an eslint expert :) Thank you!

Only thing I know is that (1) I use airbnb with eslint as documented (2) the rule `import/no-useless-path-segments` is set to `error` and (3) it doesn't fire when I expect it to.

**Comment by ljharb** [comment]
@simlu `"import/no-useless-path-segments": [2, { "commonjs": true }]`

Since the airbnb config expects you're not using require anywhere, only `import`, your expectation only holds for `import`.

**Comment by simlu** [comment]
@ljharb Ah!! Finally I understand what is going on. What a journey to get here :)

I'll do follow ups as necessary now.

Thank you very much for all the assist

**Comment by simlu** [comment]
> Separately, please do file an issue or PR to document the option in eslint-plugin-import :-)

Done [here](https://github.com/benmosher/eslint-plugin-import/issues/1459)

This is not an eslint issue, so closing.
